frontend/src/components/ViewTrips/create-new-trip.js

```diff
@@ -96,26 +96,36 @@ function formatTripData(rawTripObj) {
  * Promise containing a reference to the newly created document.
  *
  * @param {firebase.firestore.Firestore} db The Firestore database instance.
+ * @param {?string} tripId Document id associated with trip to overwrite.
+ *     Null if adding a new trip.
  * @param {Object} tripObj A JS Object containing the Trip document fields.
  */
-export function addTripToFirestore(db, tripObj) {
-  return db.collection(COLLECTION_TRIPS)
-    .add(tripObj)
+export function addTripToFirestore(db, tripId, tripObj) {
+  if (tripId === null) {
+    return db.collection(COLLECTION_TRIPS).add(tripObj);
+  }
+  return db.collection(COLLECTION_TRIPS).doc(tripId).set(tripObj);
```

Comment:
In what instance would the tripId already exist? How do we get to this line?

Reply:
If you're trying to update a trip, try `.doc().update(tripObj)`. It changes the fields provided without erasing any of the other fields.

Reply:
So they way I envisioned the 'Edit Trip' modal is similar to the Add Trip modal except each text box is populated with the current values of the trip rather than being blank with the placeholder input values. So if there were 10 collaborators, rather than having to re-enter them all, they can choose to delete one and add two and keep the other nine the same. With this model, it doesn't really matter whether I overwrite with `set` or update with `update` because I will get the same result (and I doubt there is a significant performance difference. And because I see there are a few complications with update (such as updating arrays), I think it makes sense for me to stay with using `set`.

If you disagree just let me know and we can chat about it.